web-ide/ide-server/Dockerfile

```diff
@@ -1,32 +1,43 @@
-ARG DAML_VERSION=0.12.16
+FROM openjdk:8-slim
 
-FROM "digitalasset/daml-sdk:${DAML_VERSION}-master"
+ARG DAML_VERSION=0.12.21
```

Comment:
Can the bits we need to change regularly like the DAML version remain at the very top of the file?

Reply:
this is a dockerism where variables are undefined after FROM keyword.

Reply:
Why does it need to be an ARG? Why not make it an ENV, and move the CODESERVER ones up here too?

Reply:
yea...I had moved them up...I swear I've been seeing outdated code and comments more often in github.

Saw your comment too late but having them set ENV variable implies to me that they are used inside the container, which they aren't. I don't really feel strongly about it though...